### ISSUE TITLE
RX-S601D no access to the media folder

### ISSUE THREAD

**Issue by CHWurst** [mvdwetering/yamaha_ynca#213]
**Describe the bug**
I cannot select or search for any media using the controls or Media in the menu. Only the last track or the last radio station (Net Radio) is played. When I create an automation and use "Play Media" as the action, I get the error message "The media player does not support media browsing.". But with the HA integration or the MusicCast integration it works.

**Expected behavior**
Selection of media via the control card.

**Diagnostics**
[config_entry-yamaha_ynca-de71815ec43ebd870cf0680b4ba48c3f.json.txt](https://github.com/mvdwetering/yamaha_ynca/files/13225127/config_entry-yamaha_ynca-de71815ec43ebd870cf0680b4ba48c3f.json.txt)

**Logging**
No

**Environment**

- Home Assistant Core version: 2023.10.5
- Home Assistant operating environment (OS/Container/Supervised/Core): OS via Vm on Windows 10
- Huesyncbox integration version: 
- Last working version (if known): 

**Additional context**

**Comment by mvdwetering** [comment]
Indeed, there is currently no support to browse or play media.

I had a quick look and it seems a bit complicated as from the receiver I can basically only navigate the menu pages (I guess it will also show on screen). Next to that HA does not seem to support paged browsing. I guess I could get all the pages first, but that could be slow.

For which sources are you looking at this functionality?
If the source also support Presets it would be easier to implement.

**Comment by CHWurst** [comment]
Hey Michel,

Ok, that explains why it doesn't work. Then it's more of a new feature and not a bug. I need this for the Server and Net Radio sources. But if it's too difficult then don't do it, but if it's a challenge for you then go for it. 🤣🤣😉😉



Von meinem/meiner Galaxy gesendet



-------- Ursprüngliche Nachricht --------
Von: Michel van de Wetering ***@***.***>
Datum: 04.11.23 10:16 (GMT+01:00)
An: mvdwetering/yamaha_ynca ***@***.***>
Cc: CHWurst ***@***.***>, Author ***@***.***>
Betreff: Re: [mvdwetering/yamaha_ynca] RX-S601D no access to the media folder (Issue #213)


Indeed, there is currently no support to browse or play media.

I had a quick look and it seems a bit complicated as from the receiver I can basically only navigate the menu pages (I guess it will also show on screen). Next to that HA does not seem to support paged browsing. I guess I could get all the pages first, but that could be slow.

For which sources are you looking at this functionality?
If the source also support Presets it would be easier to implement.

**Comment by mvdwetering** [comment]
I assume that Server is for accessing DLNA servers (on my receiver that input is called PC). In that case maybe the DLNA Digital Media Renderer integration in HA works for you. It does not work with my receiver, but that is a pretty old one it might work on other models.

I might give it a go anyway and see how well it works.

**Comment by CHWurst** [comment]
Yes, that's right, the input server accesses DLNA. I first switched back to the standard Yamaha integration so that at least the server and net radio inputs work. 😉

**Comment by mvdwetering** [comment]
FYI I have been experimenting with browsing the library through the YNCA protocol, but having some issues getting it stable. Response speed depends heavily on the used input e.g. USB seems very slow, DLNA is pretty fast. Next to that sometimes my receiver hangs after getting a bunch of pages and I need to reconnect. 
So unfortunately not ready to include in a release right now. I will try to get back to this at some point.


However I noticed from your logs that the receiver seems to support SONG and ALBUM for the NETRADIO source. I think it would be a nice addition for the integration, but I need some additional info to be see if it works like for the other sources.

Could you try to send the following commands to the receiver?
These commands will not change any settings, just request some data.

```
@NETRADIO:METAINFO=?
@SYS:MODELNAME=?
@NETRADIO:SONG=?
@NETRADIO:ALBUM=?
@NETRADIO:ARTIST=?
```

Sending the commands can be done in the Home Assistant developer tools on the "services" tab use the "Yamaha (YNCA): Send raw YNCA command" service to send the commands above. These can be pasted all at once into the "Raw YNCA data" field. Make sure to select your device as target before clicking "Call service"

Afterwards please upload the diagnostics to I can look at the responses from the receiver.
